src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSyncController.scala

```diff
@@ -0,0 +1,195 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor._
+import io.iohk.ethereum.blockchain.sync.SyncController.StartRegularSync
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.network.PeerActor.Status.{Chain, Handshaked}
+import io.iohk.ethereum.network.PeerActor._
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.utils.Config
+import org.spongycastle.util.encoders.Hex
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.Try
+
+trait RegularSyncController {
+  self: SyncController =>
+
+  private var headersQueue: Seq[BlockHeader] = Seq.empty
+  private var resolvingBranch = false
+  private var broadcasting = false
+  private var timeout: Option[Cancellable] = None
+
+  import Config.FastSync._
+
+  def regularSync(): Receive = {
+
+    case StartRegularSync =>
+      askForHeaders()
+
+    case ResumeRegularSync =>
+      askForHeaders()
+
+    case MessageReceived(m: BlockHeaders) if !resolvingBranch =>
+      cancelTimeout()
+      handleDownload(m)
+
+    case MessageReceived(m: BlockHeaders) if resolvingBranch =>
```

Comment:
they should, and we are checking that:
  https://github.com/input-output-hk/etc-client/pull/77/files#diff-6d6971fddeafb58d207e230d258a58c0R185